homotopy-core/src/contraction.rs

```diff
@@ -108,9 +108,12 @@ impl DiagramN {
                 signature,
             )?;
 
-            Ok::<_, ContractionError>(vec![cospan])
-        })
-        .or(Err(ContractionError::Invalid))
+            Ok(vec![cospan])
+        }) {
+            Ok(d) => Ok(d),
+            Err(either::Either::Left(e)) => Err(e),
+            Err(either::Either::Right(e)) => panic!("Attach in contract failed due to a dimension error. Please open an issue on GitHub with the action dump. Error: {}", e),
```

Comment:
That's a bit too verbose and it feels weird to tell people to report this on GitHub in the panic message. I would go with something like "boundary path in contraction was invalid".